tasmota/tasmota_xdrv_driver/xdrv_48_timeprop.ino

```diff
@@ -135,19 +166,103 @@ void TimepropSetPower(int index, float power) {
 }
 
 void TimepropInit(void) {
-  // AddLog(LOG_LEVEL_INFO, PSTR("TPR: Timeprop Init"));
-  int cycleTimes[TIMEPROP_NUM_OUTPUTS] = {TIMEPROP_CYCLETIMES};
-  int deadTimes[TIMEPROP_NUM_OUTPUTS] = {TIMEPROP_DEADTIMES};
-  int opInverts[TIMEPROP_NUM_OUTPUTS] = {TIMEPROP_OPINVERTS};
-  int fallbacks[TIMEPROP_NUM_OUTPUTS] = {TIMEPROP_FALLBACK_POWERS};
-  int maxIntervals[TIMEPROP_NUM_OUTPUTS] = {TIMEPROP_MAX_UPDATE_INTERVALS};
-
   for (int i = 0; i < TIMEPROP_NUM_OUTPUTS; i++) {
     Tprop.timeprops[i].initialise(cycleTimes[i], deadTimes[i], opInverts[i], fallbacks[i],
       maxIntervals[i], Tprop.current_time_secs);
   }
 }
 
+void CmndSetPower(void) {
+  float newPower=CharToFloat(XdrvMailbox.data);
+  if (XdrvMailbox.index >=0 && XdrvMailbox.index < TIMEPROP_NUM_OUTPUTS && newPower>=0.0 && newPower<=1.0) {
+    timeprops[XdrvMailbox.index].setPower(newPower, Tprop.current_time_secs );
+    Response_P(PSTR("{\"" D_CMND_TIMEPROP_PREFIX D_CMND_TIMEPROP_SETPOWER "%d\":\"%.2f\"}"), XdrvMailbox.index, newPower);
+  }
+}
+
+// commands for settings all take the same form
+// prefix commands with 'timeprop'
+// then append the command string eg. 'SetCycleTime'
+// then append the output number (0 for a single output)
+// then append the value to set set, or
+// leave blank to retrieve the current value
+// eg.
+// 'TimepropSetCycleTime0 120' will set the value of cycle time for output 0 to 120
+// 'TimepropSetCycleTime0' will retrieve the value of cycle time for output 0
+
+void CmndSetCycleTime(void) {
+  if (XdrvMailbox.index >=0 && XdrvMailbox.index < TIMEPROP_NUM_OUTPUTS ) {
+    if(XdrvMailbox.data_len) {
+      int newCycleTime=TextToInt(XdrvMailbox.data);
+      if(newCycleTime>0) {
+        cycleTimes[XdrvMailbox.index] = newCycleTime;
+        Tprop.timeprops[XdrvMailbox.index].initialise(cycleTimes[XdrvMailbox.index], deadTimes[XdrvMailbox.index], opInverts[XdrvMailbox.index], fallbacks[XdrvMailbox.index], maxIntervals[XdrvMailbox.index], Tprop.current_time_secs);
+        Response_P(PSTR("{\"" D_CMND_TIMEPROP_PREFIX D_CMND_TIMEPROP_SETCYCLETIME "%d\":\"%d\"}"), XdrvMailbox.index, newCycleTime);
```

Comment:
Pls verify as JSON values should not be using quotes.

Reply:
I did and it looks valid to me. I also put various output through a json validator.

Here's an example output
```
21:43:44.959 MQT: tele/tasmota_E7AD3C/SENSOR = {"Time":"2023-08-14T21:43:44","HDC1080":{"Temperature":22.6,"Humidity":100.0,"DewPoint":22.6},"ESP32":{"Temperature":26.1},"Timeprop":{"Output0":{"CycleTime":600,"DeadTime":0,"OutputInvert":0,"FallbackPower":0.00,"MaxUpdateInterval":120}},"PID":{"PidPv":22.70,"PidSp":9.00,"PidPb":5.00,"PidTi":1800.00,"PidTd":15.00,"PidInitialInt":0.50,"PidDSmooth":3.00,"PidAuto":1,"PidManualPower":0.00,"PidMaxInterval":300,"PidInterval":10,"PidUpdateSecs":0,"PidSensorLost":0,"PidPower":0.00},"TempUnit":"C"}
```

Reply:
You'll have to agree this is wrong:
```
Response_P(PSTR("{\"" D_CMND_TIMEPROP_PREFIX D_CMND_TIMEPROP_SETCYCLETIME "%d\":\"%d\"}"), XdrvMailbox.index, newCycleTime);
```
as it ends up like this:
```
{" D_CMND_TIMEPROP_PREFIX D_CMND_TIMEPROP_SETCYCLETIME 1":"123"}
```
The last quotes shouldn't be there as it's a number, not a string.

Reply:
Ah, this is the json output as a result of the commands. I used the same code as was in the single command there previously for the new commands added. This code was always wrong, so I've now replaced with `ResponseCmndFloat` etc.

Sorry about that. I hadn't spotted and/or realised.